[BUG] test_col_size_exceeding_cudf_limit fails on [databricks]

**Describe the bug**

```bash
=========================== short test summary info ============================
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[single_partition_single_empty_value_with_nulls-parquet][DATAGEN_SEED=1742852887, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 3378
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[single_partition_single_value_without_nulls-parquet][DATAGEN_SEED=1742852887, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 10
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[single_partition_multiple_value-parquet][DATAGEN_SEED=1742852887, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 1686
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_single_value-parquet][DATAGEN_SEED=1742852887, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 1364
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_int_value-parquet][DATAGEN_SEED=1742852887, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 2581
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_multiple_value_wider_first_col-parquet][DATAGEN_SEED=1742852887, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 706
FAILED ../../src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_multiple_value_narrow_first_col-parquet][DATAGEN_SEED=1742852887, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 706
============= 7 failed, 11 passed, 10 warnings in 72.93s (0:01:12) =============
```

It appears all these test items have the same assert violation
```
AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 2543
```


It does not appear DATAGEN_SEED dependent




**Steps/Code to reproduce bug**
```
TEST_MODE=CI_PART1 TESTS=col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit ./jenkins/databricks/test.sh
```

**Expected behavior**

**Environment details (please complete the following information)**

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala
@@ -1111,11 +1111,7 @@ val GPU_COREDUMP_PIPE_PATTERN = conf("spark.rapids.gpu.coreDump.pipePattern")
     .stringConf
     .transform(_.toUpperCase(java.util.Locale.ROOT))
     .checkValues(RapidsReaderType.values.map(_.toString))
-    .createWithDefault(
-      ShimLoader.getShimVersion match {
-        case DatabricksShimVersion(3, 5, 0, "14.3") => RapidsReaderType.PERFILE.toString
-        case _ => RapidsReaderType.AUTO.toString
-      })
+    .createWithDefault(RapidsReaderType.AUTO.toString)
 
   val PARQUET_DECOMPRESS_CPU =
     conf("spark.rapids.sql.format.parquet.decompressCpu")

